include/openenclave/internal/plugin.h

```diff
@@ -0,0 +1,465 @@
+        const void* opt_params,
+        size_t opt_params_size,
+        uint8_t** report_buffer,
+        size_t* report_buffer_size);
+
+    /**
+     * Frees the generated attestation evidence and endorsements.
+     *
+     * @experimental
+     *
+     * @param[in] context A pointer to the attester plugin struct.
+     * @param[in] evidence_buffer A pointer to the evidence buffer.
+     * @retval OE_OK on success.
+     * @retval An appropriate error code on failure.
+     */
+    oe_result_t (
+        *free_evidence)(oe_attester_t* context, uint8_t* evidence_buffer);
+
+    /**
+     * Frees the generated attestation endorsements.
+     *
+     * @experimental
+     *
+     * @param[in] context A pointer to the attester plugin struct.
+     * @param[in] endorsements_buffer A pointer to the endorsements buffer.
+     * @retval OE_OK on success.
+     * @retval An appropriate error code on failure.
+     */
+    oe_result_t (*free_endorsements)(
+        oe_attester_t* context,
+        uint8_t* endorsements_buffer);
+};
+
+#endif // OE_BUILD_ENCLAVE
+
+/**
+ * The verifier attestion role. The verifier is reponsible for verifying the
+ * attestation evidence and must implement the functions below.
+ */
+typedef struct _oe_verifier oe_verifier_t;
+struct _oe_verifier
+{
+    /**
+     * The base attestation role containing the common functions for each role.
+     */
+    oe_attestation_role_t base;
+
+    /**
+     *
+     * Gets the optional settings data for thhe input evidence format.
+     *
+     * @experimental
+     *
+     * @param[in] context A pointer to the verifier plugin struct.
+     * @param[in] format The format for which to retrieve the optional settings.
+     * @param[out] settings An output pointer that will be assigned the address
+     * of a dynamically allocated buffer that holds the returned settings. This
+     * pointer can be assigned a NULL value if there is no settings needed.
+     * @param[out] settings_size A pointer that points to the size of the
+     * returned format settings buffer (number of bytes).
+     * @retval OE_OK on success.
+     * @retval OE_INVALID_PARAMTER At least one of the parameters is invalid.
+     * @retval other appropriate error code.
+     */
+    oe_result_t (*get_format_settings)(
+        oe_verifier_t* context,
+        uint8_t** settings,
+        size_t* settings_size);
+
+    /**
+     * Verifies the attestation evidence and returns the claims contained in
+     * the evidence.
+     *
+     * Each plugin must return the following required claims:
+     *  - id_version (uint32_t)
+     *      - Version number. Must be 1.
+     *  - security_version (uint32_t)
+     *      - Security version of the enclave. (ISVN for SGX).
+     * - attributes (uint64_t)
+     *      - Attributes flags for the evidence:
+     *          - OE_REPORT_ATTRIBUTES_DEBUG: The evidence is for a debug
+     * enclave.
+     *          - OE_REPORT_ATTRIBUTES_REMOTE: The evidence can be used for
```

Comment:
This file is not SGX specific but the term "REPORT" is SGX specific so these should be renamed.

Reply:
These are names in existing code, referenced in many code files and documents.